Scripts/Player/Player.gd

```diff
@@ -70,6 +70,16 @@ func _process(delta):
 		position.y -= delta * MoveSpeed
 	if Input.is_action_pressed("ui_down"):
 		position.y += delta * MoveSpeed
+	
+	if Input.is_action_just_pressed("escape"):
+		if !game_menu.visible:
```

Comment:
I prefer we do positives first.
i.e. if game_menu.visible:
// if visible code
else:
// if not visible code